backend/app/views.py

```
@@ -1970,34 +1978,77 @@ def tool_settings(request, instance):
         logger.error(f"Error getting/creating settings for '{instance}': {e}", exc_info=True)
         return HttpResponse(status=500, headers={
             "HX-Trigger-After-Settle": json.dumps({"showDangerAlert": "Error loading tool settings."})})
+
+    # Check prerequisites
+    site_settings, _ = SiteSettings.objects.get_or_create(pk=1)
```

Ruff (0.14.0)

`SiteSettings` may be undefined, or defined from star imports

(F405)